app/routes.py

```diff
@@ -1 +1,245 @@
+    else:
+        tasks = Task.query.all()
+    for task in tasks:
+        if not task.completed_at:
+            task_response.append({"id":task.task_id,
+            "title":task.title,
+            "description": task.description,
+            "is_complete": False
+
+            })
+        else:
+            task_response.append({
+            "id":task.task_id,
+            "title":task.title,
+            "description": task.description,
+            "completed_at":task.completed_at
+        })
+    return jsonify(task_response)
+
+#read all goals
+@goals_bp.route("", methods=["GET"]) 
+def read_all_goals():
+    goal_response = []
+    goals = Goal.query.all()
+
+    for goal in goals:
+        if not goals:
+            return jsonify(goal_response)
+        else:
+            goal_response.append({
+            "id":goal.goal_id,
+            "title":goal.title})
+
+    return jsonify(goal_response)
+#read one task/ read if empty task. 
+@task_list_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    task = validate_task(task_id)
+    # goal = validate_goal(goal_id)
+    
+    if task.goal_id:
+        return{"task": {
+            "id":task.task_id,
+            "goal_id": task.goal_id,
+            "title":task.title,
+            "description":task.description,
+            "is_complete": True if task.completed_at else False}}
+    else:
+        return task.to_dict()
+#read one goal 
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    goal = validate_goal(goal_id)
+
+
+    return jsonify({"goal":{"id": goal.goal_id,
+        "title":goal.title}})
+# #update task
+@task_list_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+
+    task = validate_task(task_id)
+    request_body = request.get_json()
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    db.session.commit()
+    return task.to_dict()
+
+#update goal
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_goal(goal_id)
+    request_body = request.get_json()
+    goal.title = request_body["title"]
+    db.session.commit
+    return goal.to_dict_goal()
+
+#mark_complete endpoint with slack api implementation
+@task_list_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def update_task_to_complete(task_id):
+    task = validate_task(task_id)
+    task.completed_at = datetime.now()
+    #slack implementation
+    url = "https://slack.com/api/chat.postMessage"
+    payload = json.dumps({
+    "channel": "C0581AUJACV",
+    "text": (f"Someone just completed the task {task.title}")
+    })
+    headers = {
+    'Authorization': os.environ.get("SLACK_API_TOKEN"),
+    'Content-Type': 'application/json'
+    }
+    response = requests.request("POST", url, headers=headers, data=payload)
+    print(response.text)
+    db.session.commit()
+    return task.to_dict(), 200
```

Comment:
This function is pretty packed, for readability I suggest using more spacing in between lines. Something like this:
```
def update_task_to_complete(task_id):
    task = validate_task(task_id)
    task.completed_at = datetime.now()
    
    #slack implementation
    url = "https://slack.com/api/chat.postMessage"

    payload = json.dumps({
        "channel": "C0581AUJACV",
        "text": (f"Someone just completed the task {task.title}")
    })

    headers = {
        'Authorization': os.environ.get("SLACK_API_TOKEN"),
        'Content-Type': 'application/json'
    }

    response = requests.request("POST", url, headers=headers, data=payload)
    print(response.text)

    db.session.commit()

    return task.to_dict(), 200
```